core/src/history/events.rs

```diff
@@ -199,7 +232,46 @@ impl StableXOrderBookReading for EventRegistry {
         &'a self,
         batch_id_to_solve: u32,
     ) -> BoxFuture<'a, Result<(AccountState, Vec<Order>)>> {
-        async move { self.auction_state_for_batch(batch_id_to_solve) }.boxed()
+        immediate!(self.auction_state_for_batch(batch_id_to_solve))
+    }
+}
+
+impl HistoricOrderbookReading for EventRegistry {
+    fn get_auction_data_for_block<'a>(
+        &'a self,
+        block_number: BlockNumber,
```

Comment:
Maybe it makes sense to change the trait so that it is always a u64 and cannot be Earliest, Latest, Pending since those values probably do not make sense to be used. Earliest is always empty, if you want latest or pending you wouldn't use the **Historic**OrderbookReading.